### DISCUSSION TITLE
❓ How to setup max speech duration?

### DISCUSSION THREAD

**Discussion by garymmi** [snakers4/silero-vad#42] [Q&A] [answered]
as title
use default vad parameters to do speech segmentation
some duration of segments are over 3 minutes
any parameter as max_speech_duration to set ?

start=  0.00, dur=  0.94
start=  2.75, dur=  9.88
start= 13.50, dur=  6.56
start= 20.75, dur=  3.06
start= 23.88, dur= 41.44
start= 66.62, dur=  1.06
start= 69.56, dur=144.50
start=214.31, dur= 61.56
start=276.12, dur=257.50
start=533.62, dur= 27.00
start=560.81, dur= 54.19
start=614.94, dur=213.44
start=828.31, dur=  6.44

thanks

**Comment by snakers4** [comment, marked as answer]
Hi,

Have you checked, is this really long non-stop speech (or two people talking together) and not some wrong hyper-parameter choice?

In case of really long speech there is no clear obvious solution.
Because from a standpoint of our VAD the speech does not really end.

Together with STT a proper solution would be:

- Find a long speech chunk;
- Just divide in into N equal parts (or into N-second chunks);
- Get timestamps of words in each chunk;
- Re-cut chunks not to cut any words "in the middle";

**Comment by garymmi** [comment]
the speech is two people talking together, free talk
